tests/bootstrap.php

```diff
@@ -180,6 +186,26 @@ public static function instance() {
 
 		return self::$instance;
 	}
+
+	/**
+	 * Replace LegacyProxy to MockableLegacyProxy from the WC container.
+	 *
+	 * @throws \Exception Thrown when reflection fails.
+	 */
+	private function replace_legacy_proxy() {
+		try {
+			$inner_container_property = new \ReflectionProperty( \Automattic\WooCommerce\Container::class, 'container' );
+		} catch ( ReflectionException $ex ) {
+			throw new \Exception( "Error when trying to get the private 'container' property from the " . \Automattic\WooCommerce\Container::class . ' class using reflection during unit testing bootstrap, has the property been removed or renamed?' );
+		}
+
+		$inner_container_property->setAccessible( true );
+		$inner_container = $inner_container_property->getValue( wc_get_container() );
+
+		$inner_container->replace( LegacyProxy::class, MockableLegacyProxy::class );
+		$inner_container->reset_all_resolved();
+
```

Comment:
Should we also add this line? https://github.com/woocommerce/woocommerce/blob/trunk/tests/legacy/bootstrap.php#L154 similar to what woo core does, this is a little over my head, and given this fixes the issue I am fine with keeping it out for now.

Reply:
I think It's going to be tricky to check for `reset()` method due to how IoC container resolves the class. It will always return `undefined` unless we replace LegacyProxy with mockableLegacyProxy since LegacyProxy doesn't have the method. 

Adding `$GLOBALS['wc_container'] = $inner_container;` sounds good. Updated in https://github.com/woocommerce/woocommerce-admin/pull/7404/commits/914e27875e624700009068388281b55f3a2096ee

Reply:
> I think It's going to be tricky to check for reset() method due to how IoC container resolves the class. It will always return undefined unless we replace LegacyProxy with mockableLegacyProxy since LegacyProxy doesn't have the method.

Sounds good, we can leave it as is, thanks for adding the extra line 😄